app/library/postgres.py

```diff
@@ -102,9 +102,22 @@ async def get_assets_from_postgres(session):
 
 
 async def delete_asset_from_postgres(asset_id: str, session):
-    if not delete_asset(asset_id, session):
+    if not get_asset(asset_id, session):
         raise ValueError("Asset ID not valid")
 
+    scenes_resp = await get_scenes_from_postgres(session)
+    for scene in scenes_resp["scenes"]:
+        if asset_id == str(scene["background_id"]):
```

Comment:
We should have 2 for loops, one for this and after for the deletion (that's outside this for loop), otherwise we may delete half objects, then error.
Like if scene 1 and 3 have the asset as object, and scene 2 has it as background. Then, scene 1's objects will delete, but scene 3's won't because we errored halfway through (when checking scene 2).